source/cell/cell.cpp

```diff
@@ -276,6 +283,46 @@ void cell::value(const cell c)
     d_->format_ = c.d_->format_;
 }
 
+void cell::copy_from_other_workbook(const cell &source)
+{
+    d_->type_ = source.d_->type_;
+
+    // Handle shared_string: remap to destination workbook
+    if (source.data_type() == type::shared_string)
+    {
+        const auto text = source.value<rich_text>();
+        const auto new_index = workbook().add_shared_string(text, false);
```

Comment:
maybe we can just call `cell::value(const rich_text &text)` to avoid duplicating those internal details. Or implement it in a separate function if you want to avoid the unnecessary call to `check_string` (called by `cell::value(const rich_text &text)`)